Merge stagenet to development DO NOT SQUASH

stringhandler opened the pull request.

SWvheerden committed:
feat: change the miner default grpc port (#5199)

Description
---
Changes the default miner port to 18172/18173. This was the 18142/18143 which was the default Esmeralda network. 
The default network is now stagenet and not Esmeralda. This makes it that the miner works with the default base-node/wallet version out of the box

How Has This Been Tested?
---
manual

Commit:
v0.45.1-stagenet.0

brianp commented:
I actually thing we should _not_ do this.

Thought about it this morning. The appropriate way to handle the hotfix is:
Merge fix into stagenet branch
Also merge fix into development branch

Hotfixes will always require two PR's. MainNet, StageNet, nor NextNet should ever get merged backwards into development.

brianp reviewed (changes requested):
I actually think we should not do this.

Thought about it this morning. The appropriate way to handle the hotfix is:
Merge fix into stagenet branch
Also merge fix into development branch

Hotfixes will always require two PR's. MainNet, StageNet, nor NextNet should ever get merged backwards into development.

SWvheerden committed:
fix: peers stagenet (#5203)

Description
---
This updates the peer seeds that are shipped with the base_node to ensure it's shipped with the correct ones being the nodes that actually run.

Commit:
chore(release): v0.48.0-rc.0

Commit:
chore(release): v0.48.0-rc.1

Commit:
Merge branch 'st-nextnet-release' into nextnet

# Conflicts:
#	Cargo.lock
#	applications/tari_app_grpc/Cargo.toml
#	applications/tari_app_utilities/Cargo.toml
#	applications/tari_base_node/Cargo.toml
#	applications/tari_console_wallet/Cargo.toml
#	applications/tari_merge_mining_proxy/Cargo.toml
#	applications/tari_miner/Cargo.toml
#	base_layer/common_types/Cargo.toml
#	base_layer/core/Cargo.toml
#	base_layer/key_manager/Cargo.toml
#	base_layer/mmr/Cargo.toml
#	base_layer/p2p/Cargo.toml
#	base_layer/service_framework/Cargo.toml
#	base_layer/tari_mining_helper_ffi/Cargo.toml
#	base_layer/wallet/Cargo.toml
#	base_layer/wallet_ffi/Cargo.toml
#	common/Cargo.toml
#	common_sqlite/Cargo.toml
#	comms/core/Cargo.toml
#	comms/dht/Cargo.toml
#	comms/rpc_macros/Cargo.toml
#	infrastructure/derive/Cargo.toml
#	infrastructure/shutdown/Cargo.toml
#	infrastructure/storage/Cargo.toml
#	infrastructure/test_utils/Cargo.toml

Commit:
chore(release): v0.48.0-rc.2

brianp committed:
Merge branch 'nextnet' into stagenet-0.48.0

brianp committed:
Add new changelog for stagenet

Commit:
chore(release): stagenet to 0.48.0 (#5313)

Description
---
Rollover from 0.45.0 to 0.48.0. Moving NextNet to StageNet.

Motivation and Context
---
Releasing features and keeping things stable-ish.

How Has This Been Tested?
---
It's been running on NextNet.

What process can a PR reviewer use to test or verify this change?
---


Breaking Changes
---
This has multiple non backwards-compat breaking changes, but should be
the last set of non backwards compat breaking changes we have.

- [ ] None
- [x] Requires data directory on base node to be deleted
- [x] Requires hard fork
- [x] Other - Please specify

<!-- Does this include a breaking change? If so, include this line as a
footer -->
<!-- BREAKING CHANGE: Description what the user should do, e.g. delete a
database, resync the chain -->

SWvheerden committed:
chore: update location of compiled bins in s3 for stagenet (#5366)

Separate the bins by network type to make it easier to distinguish on
the tari website for download.

Motivation and Context
---
It makes it easier to parse on the website, allowing us to deploy the
new design which should help people disambiguate which binary they
should download.

This was updated for TestNet, but not for Stagenet
